fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneDocumentFromRecord.java

```diff
@@ -193,12 +193,14 @@ public void addField(@Nonnull T source, @Nonnull final String fieldName, @Nullab
         private final Object value;
         private final LuceneIndexExpressions.DocumentFieldType type;
         private final boolean stored;
+        private final Map<String, Object> fieldConfigs;
 
-        public DocumentField(@Nonnull String fieldName, @Nullable Object value, LuceneIndexExpressions.DocumentFieldType type, boolean stored) {
+        public DocumentField(@Nonnull String fieldName, @Nullable Object value, LuceneIndexExpressions.DocumentFieldType type, boolean stored, @Nullable Map<String, Object> fieldConfigs) {
             this.fieldName = fieldName;
             this.value = value;
             this.type = type;
             this.stored = stored;
+            this.fieldConfigs = fieldConfigs;
```

Comment:
I think a `null` value of `fieldConfigs` is semantically identical to the empty map (`Collections.emptyMap()`), so it might make sense to initialize the map to empty instead of `null`, and then I think you can save some null checks.

Reply:
Sounds good. Use ```Collections.emptyMap()``` now.